ios/RNEngageBySailthru.m

```diff
@@ -18,22 +18,13 @@ @implementation RNEngageBySailthru
 
 RCT_EXPORT_MODULE();
 
-- (instancetype)init {
-    self = [super init];
+- (RNEngageBySailthru *)initializeEngageBySailthruWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject {
```

Comment:
What I was proposing was something like:
```
- (EngageBySailthru *)engageBySailthruWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject: {
    NSError *error;
    EngageBySailthru *engageBySailthru = [[EngageBySailthru alloc] initWithError:&error];
    if (error) {
        [RNEngageBySailthru rejectPromise:reject withError:error];
        return nil;
    }
    return engageBySailthru;
}
```
Then in each `RCT_EXPORT_METHOD` method, you can use:
```
[[self engageBySailthruWithResolver:resolve rejecter:reject] <method here>];
```
or check for nil and return early, to avoid unnecessary work.

You can probably remove the `RNEngageBySailthru` init method entirely, since it doesn't have anything to do anymore.